app/assets/javascripts/oxalis/view/layouting/layout_persistence.js

```diff
@@ -18,56 +24,137 @@ const localStorageKeys = {
-  clearStoredLayouts();
+layoutEmitter.on("resetLayout", (layoutKey: LayoutKeys, activeLayout: string) => {
+  storeLayoutConfig(defaultLayouts[layoutKey], layoutKey, activeLayout);
 });
 
 const persistLayoutConfigsDebounced = _.debounce(persistLayoutConfigs, 1000);
 
-export function getLayoutConfig(layoutKey: LayoutKeys) {
-  if (storedLayouts[layoutKey]) {
-    // Use default dimensions and settings
-    const { dimensions, settings } = defaultLayouts[layoutKey];
-    return {
-      ...storedLayouts[layoutKey],
-      dimensions,
-      settings,
-    };
+export function getLayoutConfig(layoutKey: LayoutKeys, activeLayoutName: string) {
+  const storedLayouts = Store.getState().uiInformation.storedLayouts;
+  if (!storedLayouts[layoutKey]) {
+    return defaultLayouts[layoutKey];
+  }
+  const layout = storedLayouts[layoutKey][activeLayoutName];
+  if (!layout) {
+    return defaultLayouts[layoutKey];
   }
+  // Use default dimensions and settings
+  const { dimensions, settings } = defaultLayouts[layoutKey];
+  return {
+    ...layout,
+    dimensions,
+    settings,
+  };
+}
+
+function getDeepCopyOfStoredLayouts(): Object {
+  const currentLayouts = Store.getState().uiInformation.storedLayouts;
+  return _.cloneDeep(currentLayouts);
+}
 
-  return defaultLayouts[layoutKey];
+export function storeLayoutConfig(layoutConfig: Object, layoutKey: LayoutKeys, layoutName: string) {
+  const newLayouts = getDeepCopyOfStoredLayouts();
+  if (newLayouts[layoutKey] && newLayouts[layoutKey][layoutName]) {
+    newLayouts[layoutKey][layoutName] = layoutConfig;
+  }
+  Store.dispatch(setStoredLayoutsAction(newLayouts));
+  persistLayoutConfigsDebounced();
 }
 
-export function storeLayoutConfig(layoutConfig: Object, layoutKey: string) {
-  storedLayouts[layoutKey] = layoutConfig;
+export function deleteLayout(layoutKey: LayoutKeys, layoutName: string) {
+  const newLayouts = getDeepCopyOfStoredLayouts();
+  if (newLayouts[layoutKey]) {
+    delete newLayouts[layoutKey][layoutName];
+  }
+  Store.dispatch(setStoredLayoutsAction(newLayouts));
   persistLayoutConfigsDebounced();
 }
+
+export function addNewLayout(
+  layoutKey: LayoutKeys,
+  newLayoutName: string,
+  configForLayout: Object,
+): boolean {
+  const newLayouts = getDeepCopyOfStoredLayouts();
+  if (newLayouts[layoutKey] && newLayouts[layoutKey][newLayoutName]) {
+    Toast.info("This layout name is already used.");
+    return false;
+  }
+  if (newLayouts[layoutKey]) {
+    newLayouts[layoutKey][newLayoutName] = configForLayout;
+    newLayouts[layoutKey][lastUsedLayout] = newLayoutName;
+    Store.dispatch(setStoredLayoutsAction(newLayouts));
+    persistLayoutConfigsDebounced();
+    return true;
+  }
+  Toast.warning("Could not create a new Layout.");
+  return false;
+}
+
+export function setActiveLayout(layoutKey: LayoutKeys, activeLayout: string) {
+  const newLayouts = getDeepCopyOfStoredLayouts();
+  if (newLayouts[layoutKey] && newLayouts[layoutKey][activeLayout]) {
```

Comment:
Shouldn't we throw an error if this if-check fails?

Reply:
Like an error toast that says e.g. "The selected layout was not found. Please add a layout for the selected name."?